test/coverage/CMakeLists.txt

```diff
@@ -0,0 +1,145 @@
+
+    # Run tests
+    COMMAND ${CMAKE_CTEST_COMMAND}
+
+    # Capturing lcov counters and generating report
+    COMMAND ${LCOV_COMMAND} --directory ${PROJECT_BINARY_DIR} --capture --output-file seqan3_coverage.captured
+    # merge baseline counters and captured counters
+    COMMAND ${LCOV_COMMAND} -a seqan3_coverage.baseline -a seqan3_coverage.captured --output-file seqan3_coverage.total
+    COMMAND ${LCOV_COMMAND} --remove seqan3_coverage.total ${TEST_COVERAGE_EXCLUDE_FILES} --output-file ${PROJECT_BINARY_DIR}/seqan3_coverage
+
+    BYPRODUCTS seqan3_coverage.baseline seqan3_coverage.captured seqan3_coverage.total
+
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    COMMENT "Processing code coverage counters."
+)
+set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "seqan3_coverage.baseline;seqan3_coverage.captured;seqan3_coverage.total")
+
+add_custom_target (
+    coverage ALL
+    DEPENDS
+    ${PROJECT_BINARY_DIR}/seqan3_coverage
+)
+
+add_custom_target (
+    coverage_html
+    COMMAND ${GENHTML_COMMAND} --highlight --legend --output-directory html ${PROJECT_BINARY_DIR}/seqan3_coverage
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    DEPENDS coverage
+    COMMENT "Generate coverage report."
+)
+
+add_custom_command(
+    TARGET coverage_html POST_BUILD
+    COMMAND ;
+    COMMENT "Open ${PROJECT_BINARY_DIR}/html/index.html in your browser to view the coverage report."
+)
+
+# overload seqan3_test to fit
+macro (seqan3_test target_cpp)
+    # $target_cpp = "pod_tuple_test.cpp"
+    #   * will register the global TARGET name "pod_tuple_test" and
+    #   * will register the test case name "core_pod_tuple" if
+    #     pod_tuple_test.cpp is in test/core/
+    #
+    # NOTE(marehr): ".+/test/" REGEX is greedy, that means
+    # /test/test/test/hello_test.cpp will result in an empty `test_path`
+    string (REGEX REPLACE "_test.cpp$" "" target_name ${target_cpp})
+    string (REGEX REPLACE ".+/test/" "" test_path ${CMAKE_CURRENT_LIST_DIR})
+    set (target "${target_name}_coverage_test")
+
+    add_executable (${target} ${target_cpp})
+    target_link_libraries (${target} seqan3::test::coverage)
+    add_test (NAME "${test_path}/${target}" COMMAND ${target})
+
+    # any change of a target will invalidate the coverage result;
+    # NOTE that this is a GLOBAL variable, because a normal
+    # `set(TEST_COVERAGE_ALL_TESTS)` would not propagate the result when
+    # CMakeLists.txt goes out of scope due to a `add_subdirectory`
+    set_property(GLOBAL APPEND PROPERTY TEST_COVERAGE_ALL_TESTS ${target})
+
+    unset (target_name)
+    unset (test_path)
+    unset (target)
+endmacro ()
+
+seqan3_require_ccache ()
+seqan3_require_test ()
+
+# add all unit tests
+add_subdirectories_of ("${CMAKE_CURRENT_SOURCE_DIR}/../unit")
+
+# add collected test cases as dependency
+get_property(TEST_COVERAGE_ALL_TESTS GLOBAL PROPERTY TEST_COVERAGE_ALL_TESTS)
```

Comment:
Give the target of the command a distinguishable name from the global property name. Maybe just add a underscore at the beginning.